packages/rpc-methods/src/restricted/manageState.test.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  getManageStateImplementation,
+  getValidatedParams,
+  ManageStateOperation,
+  specificationBuilder,
+} from './manageState';
+
+describe('snap_manageState', () => {
+  describe('specification', () => {
+    it('builds specification', () => {
+      const methodHooks = {
+        clearSnapState: jest.fn(),
+        getSnapState: jest.fn(),
+        updateSnapState: jest.fn(),
+      };
+
+      expect(
+        specificationBuilder({
+          allowedCaveats: null,
+          methodHooks,
+        }),
+      ).toStrictEqual({
+        allowedCaveats: null,
+        methodImplementation: expect.anything(),
+        permissionType: 'RestrictedMethod',
+        targetKey: 'snap_manageState',
+      });
+    });
+  });
+
+  describe('getManageStateImplementation', () => {
+    it('gets snap state', async () => {
+      const clearSnapState = jest.fn().mockResolvedValueOnce(true);
+      const getSnapState = jest.fn().mockResolvedValueOnce(true);
+      const updateSnapState = jest.fn().mockResolvedValueOnce(true);
+
+      const manageStateImplementation = getManageStateImplementation({
+        clearSnapState,
+        getSnapState,
+        updateSnapState,
+      });
+
+      await manageStateImplementation({
+        context: { origin: 'snap-origin' },
+        method: '',
+        params: { operation: ManageStateOperation.getState },
+      });
+
+      expect(getSnapState).toHaveBeenCalledWith('snap-origin');
```

Comment:
Can we check the return value as well here?

Reply:
Yes, checked.